redpanda/templates/configmap.yaml

```diff
@@ -26,25 +26,109 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
     app.kubernetes.io/component: {{ template "redpanda.name" . }}
-  {{- with .Values.labels }}
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+{{- with .Values.commonLabels }}
+  {{- toYaml . | nindent 4 }}
+{{- end }}
 data:
   redpanda.yaml: |
     config_file: /etc/redpanda/redpanda.yaml
-    license_key: {{ .Values.config.license_key }}
-    organization: {{ .Values.config.organization }}
+    {{- if gt (len .Values.license_key) 0 }}
+    license_key: {{ .Values.license_key }}
+    {{- end }}
+    {{- if gt (len .Values.organization) 0 }}
+    organization: {{ .Values.organization }}
+    {{- end }}
     redpanda:
-{{ toYaml .Values.config.redpanda | indent 6 }}
+{{- if .Values.auth.sasl.enabled }}
+  {{- with .Values.auth.sasl.users }}
+      superusers:
+    {{ toYaml . | indent 8 }}
+  {{- end }}
+{{- end }}
+{{- if .Values.listeners.admin.enabled }}
+      admin:
+        address: {{ .Values.listeners.admin.address }}
+        name: admin
+        port: {{ .Values.listeners.admin.port }}
+  {{- $adminTlsEnabled := include "admin-tls-enabled" . }}
+  {{- if eq $adminTlsEnabled "true" }}
+      admin_api_tls:
+      - cert_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.admin.tls.cert }}/tls.crt
+        enabled: true
+        key_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.admin.tls.cert }}/tls.key
+        name: admin
+    {{- if .Values.listeners.admin.tls.requireClientAuth }}
+        require_client_auth: true
+    {{- end }}
+        truststore_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.admin.tls.cert }}/ca.crt
+  {{- end }}
+{{- else }}
+      enable_admin_api: false
+{{- end }}
+      enable_sasl: {{ .Values.auth.sasl.enabled }}
+      kafka_api_tls:
+{{- range $i, $listener := .Values.listeners.kafka.endpoints }}
+  {{- if hasKey $listener "tls" }}
+    {{- if or $listener.tls.enabled (and (empty $listener.tls.enabled) $values.auth.tls.enabled) }}
+      - cert_file: {{ printf "/etc/tls/certs/%s" $listener.tls.cert }}/tls.crt
+        enabled: true
+        key_file: {{ printf "/etc/tls/certs/%s" $listener.tls.cert }}/tls.key
+        name: {{ $listener.name }}
+        require_client_auth: {{ $listener.tls.requireClientAuth }}
+        truststore_file: {{ printf "/etc/tls/certs/%s" $listener.tls.cert }}/ca.crt
+    {{- end }}
+  {{- end }}
+{{- end }}
+{{- if hasKey .Values.listeners.rpc "tls" }}
+  {{- if or .Values.listeners.rpc.tls.enabled (and (empty .Values.listeners.rpc.tls.enabled) .Values.auth.tls.enabled) }}
+      rpc_server_tls:
+        enabled: true
+        require_client_auth: {{ .Values.listeners.rpc.tls.requireClientAuth }}
+        cert_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.rpc.tls.cert }}/tls.crt
+        key_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.rpc.tls.cert }}/tls.key
+        truststore_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.rpc.tls.cert }}/ca.crt
```

Comment:
rpc_server_tls is a list: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/config/schema.go#L52, we should append the `-`

```
      rpc_server_tls:
        - enabled: true
          require_client_auth: {{ .Values.listeners.rpc.tls.requireClientAuth }}
          cert_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.rpc.tls.cert }}/tls.crt
          key_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.rpc.tls.cert }}/tls.key
          truststore_file: {{ printf "/etc/tls/certs/%s" .Values.listeners.rpc.tls.cert }}/ca.crt
```

Reply:
See https://github.com/redpanda-data/helm-charts/pull/85#discussion_r913704226